docs/src/index.md

```diff
@@ -39,68 +84,85 @@ iterator interface.
 ## Iterator interface
 
 To offeset the limitations of the `ODE.ode` interface we implemented a
-general.  First we define an initial value problem, in our case this is
-an explicit differential equation `y'=y` with inital data `y0=[1.0]`
-given at the time `t0=0.0`.
+general.  We use the same problem as before as an example
 
-```@example iterator
+```@example iterate
 using ODE
 t0  = 0.0
 y0  = [1.0]
 F!(t,y,dy) = dy[1]=y[1]
 ode = ODE.ExplicitODE(t0,y0,F!)
 ```
 
-Note that unlike in `ODE.ode` we now have to supply an in place
-function `F!` instead of an explicit function `F`.  Now we are ready
-to produce the iterator that solvese to our problem.
+Now we have full flow control over the solver, we can analyze the
+intermediate results or interrupt the integration at any point.
 
-```@example iterator
-sol = ODE.solve(ode)
-for (t,y) in sol
+```@example iterate
+iter = ODE.iterate(ode)
+for (t,y) in iter
     @show (t,y)
     if t > 1
         break
     end
 end
 ```
 
-Note that we had to interrupt the loop because `sol` would be
-producing solutions ad infinitum (in theory, in practice we will get
-to the point where the solver won't be able to produce reasonable
-solution anymore).  To set the final integration time and other
-parameters of the integrator `integ` we can pass optional arguments to
+Note that we had to break the loop because `sol` would keep producing
+the results.  To set the final integration time and other parameters
+of the integrator `integ` we can pass optional arguments to
 `ODE.solver`.
 
-```@example iterator
-sol = ODE.solve(ode; tstop = 1)
-for (t,y) in sol
+```@example iterate
+iter = ODE.iterate(ode; tstop = 1)
+for (t,y) in iter
     @show (t,y)
 end
 ```
 
 This approach has the added benefit of the solution never exceeding
-the final time.  Apart from the time and value `(t,y)` the `ODE.solve`
-returns also the derivative, you can retrive it as the third argument
-in the returned tuple.  In the following example we use it compute the
-absolute error.
-
-```@example iterator
-sol = ODE.solve(ode; tstop = 1)
-for (t,y,dy) in sol
+the final time.  Both `ODE.iterate` and `ODE.solve` support the same
+options, so you can easily change the method of integration with the
+keyword `solver`.
+
+Apart from the time and value `(t,y)` the `ODE.solve` also returns the
+derivative, you can retrive it as the third argument in the returned
+tuple.  In the following example we use it compute the absolute
+residual error (zero in this case).
+
+```@example iterate
+iter = ODE.iterate(ode; tstop = 1)
+for (t,y,dy) in iter
```

Comment:
I would rather leave the `tstop` as an optional argument here.  In many cases you continue the integration until you hit a specific condition and you don't know when this might happen.  It would feel awkward to specify the `tstop` if you don't care about it.